src/compute/merge_sort/mod.rs

```diff
@@ -89,13 +89,22 @@ type MergeSlice = (usize, usize, usize);
 pub fn take_arrays<I: IntoIterator<Item = MergeSlice>>(
     arrays: &[&dyn Array],
     slices: I,
+    limit: Option<usize>,
 ) -> Box<dyn Array> {
     let slices = slices.into_iter();
     let len = arrays.iter().map(|array| array.len()).sum();
-    let mut growable = make_growable(arrays, false, len);
+    let limit = limit.unwrap_or(len);
+    let mut growable = make_growable(arrays, false, limit);
 
+    let mut current_len = 0;
     for (index, start, len) in slices {
-        growable.extend(index, start, len)
+        if len + current_len >= limit {
```

Comment:
I think this probably makes the code without a limit (limit=None) slower.

Reply:
This can be avoided by having two different implementations/bodies.

Reply:
done

Reply:
I would prefer to not expose a new function just for this; wouldn't it make sense to move this `if` to outside of the loop and keep the function with the `limit` argument?

Reply:
I'm afraid the *if* can't move outside of the loop because the slice is an iterator.

Reply:
If you do `slices.into_iter()` in two code blocks instead of reusing it, it shouldn't be a compile error?

Reply:
What I mean is that`slices` is lazy iterated, we don't know the item of `len` unless we call next(), so the `if` check must be inside the loop when limit is some value.

Reply:
I was thinking in something like this:

```
...
if let Some(limit) = limit {
    iterate with the limit condition
} else {
   iterate without the limit condition
}
...
growable.to_box()
```

Reply:
Sounds great.